services/streammanager/googlepubsub/googlepubsubmanager.go

```diff
@@ -0,0 +1,198 @@
+package googlepubsub
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/tidwall/gjson"
+	"google.golang.org/api/option"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type Config struct {
+	Credentials     string              `json:"credentials"`
+	ProjectId       string              `json:"projectId"`
+	EventToTopicMap []map[string]string `json:"eventToTopicMap"`
+}
+
+type pubsubClient struct {
+	Pbs          *pubsub.Client
+	EventToTopic []map[string]interface{}
```

Comment:
Looking at the code, I think this should be 
```
Topics map[string]*pubsub.Topic     # map from topic name to topic object
``` 
instead of 
```
EventToTopic []map[string]interface{}
```
Because as far as I can tell, we are accessing topic object using topic name.

Reply:
changed